plugins/fclrave/fclmanagercache.cpp

```diff
@@ -232,35 +232,58 @@ void FCLCollisionManagerInstance::InitEnvironment(const std::vector<int8_t>& exc
     pmanager->setup();
 }
 
-void FCLCollisionManagerInstance::EnsureBodies(const std::vector<KinBodyConstPtr>& vbodies) {
-    _tmpSortedBuffer.resize(0);
+void FCLCollisionManagerInstance::EnsureBodies(const std::vector<KinBodyConstPtr>& vbodies)
+{
+    // If there are no bodies to ensure, short-circuit.
     if (vbodies.empty()) {
         return;
     }
+
+    // Clear our scratch buffer
+    _tmpSortedBuffer.resize(0);
+
+    // We need to make sure we resize _vecCachedBodies to allow direct lookup by body index, which we can do the first time we encounter a non-null body.
     bool ensuredVecCachedBodies = false;
-    std::vector<CollisionObjectPtr> vcolobjs;
+
+    // Local scratch buffer of collision objects to reuse to avoid allocations
+    std::vector<CollisionObjectPtr> vCollisionObjects;
```

Comment:
@rschlaikjer can this vector be cached somewhere so we don't have to keep on recreating it?